File: backend/app/schemas/builder.py
from datetime import datetime
from pydantic import BaseModel

class BuilderBase(BaseModel):
    name: str
    company: str
    contact_person: str
    phone: str
    email: str
    address: str | None = None
    commission_rate: float = 3.0
    notes: str | None = None

class BuilderCreate(BuilderBase):
    pass

class BuilderUpdate(BaseModel):
    name: str | None = None
    company: str | None = None
    contact_person: str | None = None
    phone: str | None = None
    email: str | None = None
    address: str | None = None
    commission_rate: float | None = None
    notes: str | None = None

class BuilderResponse(BuilderBase):
    id: int
    created_at: datetime
    updated_at: datetime
    projects_count: int = 0

    class Config:
        from_attributes = True

File: frontend/src/types/builder.ts
export interface Builder {
  id: number;
  name: string;
  company: string;
  contact_person: string;
  phone: string;
  email: string;
  address?: string;
  commission_rate: number;
  notes?: string;
  projects_count: number;
  created_at: string;
  updated_at: string;
}

export interface BuilderCreateInput {
  name: string;
  company: string;
  contact_person: string;
  phone: string;
  email: string;
  address?: string;
  commission_rate: number;
  notes?: string;
}
